text/0000-light-dom-scoped-styles.md

```diff
@@ -0,0 +1,369 @@
+---
+title: Light DOM Scoped Styles
+status: DRAFTED
+created_at: 2021-05-14
+updated_at: YYYY-MM-DD
+champion: Nolan Lawson (nolanlawson), Abdulsattar Mohammed (abdulsattar)
+pr: https://github.com/salesforce/lwc-rfcs/pull/50
+---
+
+# Light DOM Scoped Styles
+
+## Summary
+
+This proposal adds a mechanism for components (including [light DOM components](./0114-light-dom.md)) to have co-located CSS files that are scoped to that component, but without relying on shadow DOM.
+
+## Basic example
+
+Since there is already `*.css` for unscoped light DOM CSS files, we add `*.scoped.css` for scoped light DOM CSS:
+
+```html
+<!-- x/foo.html -->
+<template lwc:render-mode="light">
+  <div>Hello</div>
+</template>
+```
+
+```js
+// x/foo.js
+import { LightningElement } from 'lwc'
+
+export default class Foo extends LightningElement {
+  static renderMode = 'light'
+}
+```
+
+```css
+/* x/foo.scoped.css */
+div {
+  color: green;
+}
+```
+
+The result will be:
+
+```html
+<x-app>
+  <style class="x-foo_foo" type="text/css">
+    div.x-foo_foo { color: green; }
+  </style>
+  <div class="x-foo_foo">Hello</div>
+</x-app>
+```
+
+## Motivation
+
+For ([light DOM components](./0114-light-dom.md)), the default assumption is that styles are unscoped. In other words, a light DOM component essentially just contains `<style>` tags that are inserted into the DOM _in situ_. (The actual implementation may be slightly different, but this is the basic mental model.)
+
+
+Many frameworks, however, have a concept of _scoped styles_, even without using shadow DOM. These offer good developer ergonomics, because developers can concentrate on the CSS co-located with a particular component, without worrying how that CSS might affect other components.
+
+Prior art:
+
+- [Vue: `<style scoped>`](https://vue-loader.vuejs.org/guide/scoped-css.html)
+- [Svelte: scoped styles](https://svelte.dev/docs#style)
+- [React: CSS Modules in `create-react-app`](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/)
+- [Stencil: scoped styles in light DOM](https://stenciljs.com/docs/styling#scoped-css)
+- [BEM: Block, Element, Modifier](https://en.bem.info/)
+
+Furthermore, scoped styles are useful even for shadow DOM. Consider a shadow parent component with a light child component: the parent's styles "bleed" into the child, which may be surprising for developers who are accustomed to the Vue/Svelte model.

```

Comment:
This paragraph is confusing me a bit, are you saying scoped styles would prevent the bleeding of styles?

```html
<x-app>
  <style class="x-foo_foo" type="text/css">
    div.x-foo_foo { color: green; }
  </style>
  <x-bar>
    #shadowroot
      <style>
        h2 { color: red; }
      </style> 
      <h2 class="x-foo_foo">Hello</h2>
  </x-bar>
</x-app>
```

The light DOM element with scoped styles would still render as red

Reply:
@brandonferrua I'm talking about this case:

```html
<x-shadow>
  #shadowroot
    <style type="text/css">
      div { color: green; }
    </style>
    <x-light>
      <div>Hello</div>
    </x-light>
</x-shadow>
```

In this case, the light DOM component's `<div>` will have `color: green`, which may surprise developers who think of shadow DOM styles as "scoped." (It's "scoped," but not in the same way most JS frameworks talk about scoping!)